src/etc/tidy.py

```diff
@@ -43,78 +43,184 @@ def do_license_check(name, contents):
     if not check_license(name, contents):
         report_error_name_no(name, 1, "incorrect license")
 
-
-file_names = [s for s in sys.argv[1:] if (not s.endswith("_gen.rs"))
-                                     and (not ".#" in s)]
-
 current_name = ""
 current_contents = ""
 check_tab = True
 check_cr = True
 check_linelength = True
 
+if len(sys.argv) < 2:
+    print "usage: tidy.py <src-dir>"
+    sys.exit(1)
+
+src_dir = sys.argv[1]
 
 try:
-    for line in fileinput.input(file_names,
+    count_rs = 0
+    count_py = 0
+    count_js = 0
+    count_sh = 0
+    count_pl = 0
+    count_c = 0
+    count_h = 0
+    count_other = 0
+
+    count_lines = 0
+    count_non_blank_lines = 0
+
+    def update_counts(current_name):
```

Comment:
I think that using a dictionary to track the file counts will make it easier to maintain in the long run.

```python
file_counts = {
    '.rs': 0,
    '.py': 0,
    # ...
    'other': 0
}

def update_counts(current_name):
    global file_counts
    _, ext = os.path.splitext(current_name)
    
    if ext in file_counts:
        file_counts[ext] += 1
    else:
        file_counts['other'] += 1